packages/cli/src/lib/push/parseFile.ts

```diff
@@ -15,16 +15,18 @@ import {
 import {existsSync, lstatSync, PathLike, readFileSync} from 'fs';
 import {CaseInsensitiveDocument} from './caseInsensitiveDocument';
 import {KnownKeys} from './knownKey';
-import {InvalidDocument, NotAFileError} from './validatorErrors';
+import {
+  InvalidDocument,
+  NotAFileError,
+  NotAJsonFileError,
+} from './validatorErrors';
 import {RequiredKeyValidator} from './requiredKeyValidator';
+import {readJSONSync} from 'fs-extra';
 
 export const parseAndGetDocumentBuilderFromJSONDocument = (
   documentPath: PathLike
 ) => {
-  const isAFile = isFile(documentPath);
-  if (!isAFile) {
-    throw new NotAFileError(documentPath);
-  }
+  ensureFileIntegrity(documentPath);
 
   const fileContent = JSON.parse(readFileSync(documentPath).toString()) as
     | Record<string, string>
```

Comment:
Alright, here, it's not great perf-wise: We're reading the file twice because of the `isValidJsonFile`.
The cost will be non-negligible when facing big chunky JSON.

So, I would remove the isValidJSONFile method altogether and trycatch the `filecontent` var assignment instead.
I would also throw a NotAJsonFileError only and only if this is actually what's causing the program to throw (i.e. checks on the `e` of the catch

Reply:
Good point. However, instead of checking on the `e` of the catch, why not create a method (e.g. `safeParse`) that would only parse the JSON.

That way, we are sure any error throw by this method is actually a `NotAJsonFileError`